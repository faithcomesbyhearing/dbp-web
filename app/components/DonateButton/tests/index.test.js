import React from 'react';
import renderer from 'react-test-renderer';
import { DonateButton } from '..';

jest.mock('react-intl', () => ({
	FormattedMessage: ({ defaultMessage }) => <span>{defaultMessage}</span>,
	defineMessages: (messages) => messages,
}));

describe('<DonateButton /> component', () => {
	it('should match snapshot of paper theme', () => {
		const tree = renderer.create(<DonateButton theme={'paper'} />).toJSON();
		expect(tree).toMatchSnapshot();
	});
	it('should match snapshot of red theme', () => {
		const tree = renderer.create(<DonateButton theme={'red'} />).toJSON();
		expect(tree).toMatchSnapshot();
	});
	it('should match snapshot of dark theme', () => {
		const tree = renderer.create(<DonateButton theme={'dark'} />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
