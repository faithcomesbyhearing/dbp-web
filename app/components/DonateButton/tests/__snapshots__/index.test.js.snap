// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DonateButton /> component should match snapshot of dark theme 1`] = `
<a
  className="donate-button"
  href="https://www.faithcomesbyhearing.com/donate/form/183"
  target="_blank"
>
  <span
    className="donate-svg-wrapper"
  >
    <svg>
      <use
        xlinkHref="/svglist.svg#donate_light"
        xmlnsXlink="http://www.w3.org/1999/xlink"
      />
    </svg>
  </span>
  <span>
    Donate
  </span>
</a>
`;

exports[`<DonateButton /> component should match snapshot of paper theme 1`] = `
<a
  className="donate-button"
  href="https://www.faithcomesbyhearing.com/donate/form/183"
  target="_blank"
>
  <span
    className="donate-svg-wrapper"
  >
    <svg>
      <use
        xlinkHref="/svglist.svg#donate_dark"
        xmlnsXlink="http://www.w3.org/1999/xlink"
      />
    </svg>
  </span>
  <span>
    Donate
  </span>
</a>
`;

exports[`<DonateButton /> component should match snapshot of red theme 1`] = `
<a
  className="donate-button"
  href="https://www.faithcomesbyhearing.com/donate/form/183"
  target="_blank"
>
  <span
    className="donate-svg-wrapper"
  >
    <svg>
      <use
        xlinkHref="/svglist.svg#donate_light"
        xmlnsXlink="http://www.w3.org/1999/xlink"
      />
    </svg>
  </span>
  <span>
    Donate
  </span>
</a>
`;
